src/System.Management.Automation/engine/remoting/commands/CustomShellCommands.cs

```diff
@@ -113,8 +113,17 @@ public sealed class RegisterPSSessionConfigurationCommand : PSSessionConfigurati
             }}
         }}
 
-        new-item -path WSMan:\localhost\Plugin -file ""$filepath"" -name ""$pluginName""
-        # $? is to make sure the last operation is succeeded
+        try
+        {{
+            new-item -path WSMan:\localhost\Plugin -file ""$filepath"" -name ""$pluginName""
+        }}
+        catch [System.InvalidOperationException] # WS2012/R2 WinRM has limitation where MaxConcurrentUsers can't be greater than 100
```

Comment:
WS2012R2 has PSv3 by default and I'm not sure if the Max limit was changed in WMF4 or WMF5, so I'll just mention WMF